api/src/lib.rs

```diff
@@ -245,6 +248,45 @@ impl InternedStringId {
     }
 }
 
+/// A mechanism for caching interned string IDs.
+pub struct CachedInternedStringId {
+    value: &'static str,
+    interned_string_id: AtomicUsize,
```

Comment:
Also, any reason why you're using `AtomicUsize` here. Is it because there are legit reasons/concerns regarding sharing this between threads or mostly to update in place without having to worry about the borrow checker?